pkg/networkmanager/firewall-client.js

```diff
@@ -158,6 +180,38 @@ function fetchServiceInfos(services) {
     });
 }
 
+function fetchZoneInfos(zones) {
+    let zoneNames = Array.isArray(zones) ? zones : Object.keys(zones);
```

Comment:
Does it realistically happen that this gets called on both arrays and lists? This is a bit confusing.